accepted/00000-openapi.md

```diff
@@ -0,0 +1,184 @@
+
+It is worth noting that although this custom reader will introduce new code to maintain, it will replace the existing Doclet code currently used to generate API documentation. This transition will streamline the process and align it with modern standards.
+
+
+#### C. Swagger UI
+
+The generated OpenAPI specification will be seamlessly integrated into the Uyuni web UI using Swagger UI, enabling users to interact with API endpoints directly through the interface without requiring any coding.
+
+To implement this feature, the [swagger-ui-react](https://www.npmjs.com/package/swagger-ui-react) library will be utilized. This library simplifies integration by allowing the specification to be passed directly to the SwaggerUI component, for example: `<SwaggerUI spec={uyuniSpec} />`.
+
+Given the large number of endpoints provided by the Uyuni API, the UI will display endpoints categorized by namespaces. Users will have the ability to switch between namespaces and view the corresponding endpoints through a selection interface within the web UI. This approach ensures that the interface remains navigable and user-friendly, even with a substantial volume of endpoints.
+
+Additionally, the web UI can incorporate supplementary documentation, including general descriptions, detailed instructions, and code examples, further enhancing the usability and functionality of the platform.
+
+
+### Drawbacks
+- Custom API reader maintenance.
+  - Implementing a custom OpenAPI reader for Spark APIs introduces additional maintenance overhead. Unlike the JAX-RS reader provided by Swagger Core, the custom reader will require ongoing updates to accommodate changes in the Uyuni codebase or the Swagger/OpenAPI libraries.
+- Initial Development Complexity
+  - Creating the one-time scanner tool to automate the migration of existing Javadoc-based documentation adds complexity. While this tool will simplify the process, its development and validation represent a time-intensive task.
+
+
+### Alternatives
+
+- Keep the documentation as it is, since it is stable and functional.
+  - The existing documentation, while static and less interactive, is stable and functional. Retaining it avoids the need for migration and eliminates the development and maintenance effort associated with Swagger integration. However, this approach would not address the limitations of static documentation, such as lack of interactivity and adherence to industry standards.
+
+- Keep the format for describing the documentation using JavaDocs and write a new Java Doclet to generate the OpenAPI specification by parsing the definitions in JavaDocs.
```

Comment:
I imagine we might still be obliged to have a non-interactive online documentation in parallel.
In this case, we should consider this alternative.

Can we generate such documentation from openAPI?

Reply:
That is a great question Can. We may need to have a static API documentation to integrate in the SUSE Documentation portal. @wweellddeerr can we make it with openAPI/swagger UI in build time?